test/unit/protocols/jds_line_test.dart

```diff
@@ -0,0 +1,189 @@
+import 'dart:async';
+import 'dart:io';
+
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hmi_core/hmi_core.dart';
+import 'package:hmi_networking/hmi_networking.dart';
+
+import '../helpers.dart';
+import '../test_data.dart';
+
+
+class FakeLineSocket extends Fake implements LineSocket {
+  final Stream<Uint8List> _fakeStream;
+  final bool _isConnected;
+  final Result<bool> _requestAllResult;
+  final Result<bool> _sendResult;
+
+  FakeLineSocket({
+    Stream<Uint8List> stream = const Stream.empty(),
+    bool isConnected = true, 
+    Result<bool> requestAllResult = const Result(data: true),
+    Result<bool> sendResult = const Result(data: true),
+  }) : _fakeStream = stream,
+    _isConnected = isConnected,
+    _requestAllResult = requestAllResult,
+    _sendResult = sendResult;
+
+  @override
+  Stream<Uint8List> get stream => _fakeStream;
+
+  @override
+  bool get isConnected => _isConnected;
+
+  @override
+  Result<bool> requestAll() => _requestAllResult;
+
+  @override
+  Future<Result<bool>> send(List<int> data) => Future.value(_sendResult);
+
+  @override
+  Future close() => Future.value(true);
+}
+
+void main() {
+  group('JdsLine', () {
+    late FakeLineSocket socket;
+    late JdsLine line;
+
+      test('stream', () async {
+        socket = FakeLineSocket(
+          stream: Stream.fromFuture(
+            Future.delayed(
+              const Duration(milliseconds: 100), 
+              () => encodeDataPoints(testDataPoints),
+            ),
+          ),
+          isConnected: true,
+        );
+        line = JdsLine(lineSocket: socket);
+
+        final receivedEvents = <DsDataPoint>[];
+        line.stream.listen((event) { receivedEvents.add(event); });
+
+        await Future.delayed(const Duration(milliseconds: 100));
+        
+        expect(compareDataPointCollections(receivedEvents, testDataPoints), true);
+      });
+
+      test('requestAll', () async {
+        socket = FakeLineSocket(
+          requestAllResult: const Result(data: true),
+          sendResult: const Result(data: true),
+        );
+        line = JdsLine(lineSocket: socket);
+
+        final receivedEvents = <DsDataPoint>[];
+        line.stream.listen((event) { receivedEvents.add(event); });
+
+        await Future.delayed(const Duration(milliseconds: 100));
+        final result = await line.requestAll();
+
+        await Future.delayed(const Duration(milliseconds: 100));
+        expect(result.data, true);
+        expect(receivedEvents.every((event) => event.type == DsDataType.bool && event.value == 1), true);
+      });
+
+      test('send', () async {
```

Comment:
was sent to the server the pool of various command types must be verified first of all:

Reply:
Done:
https://github.com/a-givertzman/hmi_networking/pull/3/files#diff-863531f550408fda10395d04f67ae07424fd77925eda1eb4f4e441ae8e18fdb2